storageExample.go

```diff
@@ -3,223 +3,229 @@ package main
 
 import (
 	"encoding/base64"
+	"flag"
 	"fmt"
 	"io/ioutil"
 	"math/rand"
 	"os"
 
+	"bytes"
+
 	"github.com/Azure/azure-sdk-for-go/storage"
 )
 
-type storageEnvironment string
+var (
+	accountName string
+	accountKey  string
+	blobCli     storage.BlobStorageClient
+	emulator    *bool
 
-const (
-	emulator storageEnvironment = "emulator"
-	account  storageEnvironment = "account"
+	appendBlobFile = "appendBlob.txt"
+	blockBlobFile  = "blockBlob.txt"
+	pageBlobFile   = "pageBlob.txt"
 )
 
+func init() {
+	emulator = flag.Bool("emulator", false, "use the Azure Storage Emulator")
+	flag.Parse()
+	if *emulator {
+		accountName = storage.StorageEmulatorAccountName
+		accountKey = storage.StorageEmulatorAccountKey
+	} else {
+		accountName = getEnvVarOrExit("ACCOUNT_NAME")
+		accountKey = getEnvVarOrExit("ACCOUNT_KEY")
+	}
+	client, err := storage.NewBasicClient(accountName, accountKey)
+	onErrorFail(err, "Create client failed")
+
+	blobCli = client.GetBlobService()
+
+	// delete all containers in the emulator
+	if *emulator {
+		list, err := blobCli.ListContainers(storage.ListContainersParameters{})
```

Comment:
Where are these containers persisted, locally on disk?  Do we run the risk of deleting containers that don't belong to this sample?  If so then perhaps we should only delete the ones pertaining to this sample.